packages/python-google-compute-engine/google_compute_engine/instance_setup/instance_setup.py

```diff
@@ -28,9 +28,21 @@
 from google_compute_engine import logger
 from google_compute_engine import metadata_watcher
 from google_compute_engine.boto import boto_config
+from google_compute_engine.compat import urlerror
+from google_compute_engine.compat import urlrequest
 from google_compute_engine.instance_setup import instance_config
 
 
+class PutRequest(urlrequest.Request):
+    def get_method(self):
```

Comment:
this doesn't seem necessary.. `get_method()` will return 'PUT' if there's data, and we always pass data in to this object.

Reply:
No, if you pass in data, get_method() returns a 'POST' and guest attributes needs a 'PUT'.

Reply:
>>> data = "asdfasdf"
>>> r = urllib2.Request("http://foo.com", data)
>>> r.get_method()
'POST'